File: app/src/main/java/com/blanke/mdwechat/ui/ListViewHook.java
package com.blanke.mdwechat.ui;

import android.content.Context;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.util.AttributeSet;
import android.widget.ListView;

import de.robv.android.xposed.XC_MethodHook;
import de.robv.android.xposed.callbacks.XC_LoadPackage;

import static de.robv.android.xposed.XposedHelpers.findAndHookConstructor;

/**
 * Created by blanke on 2017/8/1.
 */

public class ListViewHook extends BaseHookUi {
    @Override
    public void hook(XC_LoadPackage.LoadPackageParam lpparam) {
        //去掉黄色的 selector
        findAndHookConstructor(ListView.class, Context.class, AttributeSet.class, new XC_MethodHook() {
            @Override
            protected void afterHookedMethod(MethodHookParam param) throws Throwable {
                ListView listView = (ListView) param.thisObject;
//                log("ListView=" + listView.getClass().getName());
                ColorDrawable transDrawable = new ColorDrawable(Color.TRANSPARENT);
                listView.setSelector(transDrawable);
//                listView.setBackground(transDrawable);
            }
        });
//        findAndHookMethod(AbsListView.class, "obtainView", int.class, boolean[].class
//                , new XC_MethodHook() {
//                    @Override
//                    protected void afterHookedMethod(MethodHookParam param) throws Throwable {
//                        View view = (View) param.getResult();
//                        Context context = view.getContext();
//                        view.setBackground(getRippleDrawable(context));
//                    }
//                });
    }
}

File: app/src/main/java/com/blanke/mdwechat/ui/ListViewHook.kt
package com.blanke.mdwechat.ui

import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.widget.ListView
import com.blanke.mdwechat.config.C
import com.blanke.mdwechat.config.HookConfig
import de.robv.android.xposed.XC_MethodHook
import de.robv.android.xposed.XposedHelpers.findAndHookConstructor
import de.robv.android.xposed.callbacks.XC_LoadPackage

/**
 * Created by blanke on 2017/8/1.
 */

class ListViewHook : BaseHookUi() {
    override fun hook(lpparam: XC_LoadPackage.LoadPackageParam) {
        if (!HookConfig.isHookripple) {
            return
        }
        //去掉黄色的 selector
        findAndHookConstructor(C.ListView, C.Context, C.AttributeSet, object : XC_MethodHook() {
            @Throws(Throwable::class)
            override fun afterHookedMethod(param: XC_MethodHook.MethodHookParam?) {
                val listView = param!!.thisObject as ListView
                //                log("ListView=" + listView.getClass().getName());
                val transDrawable = ColorDrawable(Color.TRANSPARENT)
                listView.selector = transDrawable
                //                listView.setBackground(transDrawable);
            }
        })
        //        findAndHookMethod(AbsListView.class, "obtainView", int.class, boolean[].class
        //                , new XC_MethodHook() {
        //                    @Override
        //                    protected void afterHookedMethod(MethodHookParam param) throws Throwable {
        //                        View view = (View) param.getResult();
        //                        Context context = view.getContext();
        //                        view.setBackground(getDefaultRippleDrawable(context));
        //                    }
        //                });
    }
}
